frameworklauncher/src/main/java/com/microsoft/frameworklauncher/common/utils/ValueRangeUtils.java

```diff
@@ -0,0 +1,289 @@
+// Copyright (c) Microsoft Corporation
+// All rights reserved.
+//
+// MIT License
+//
```

Comment:
Seems the code is from Hadoop, pls check the License

Reply:
If code is from Hadoop, can we just use it directly? As application master is running in Hadoop, Hadoop class path maybe already included.

---
In reply to: [167155809](https://github.com/Microsoft/pai/pull/318#discussion_r167155809) [](ancestors = 167155809)

Reply:
The code is not from Hadoop, they are similar, but a lot of different.

---
In reply to: [170791912](https://github.com/Microsoft/pai/pull/318#discussion_r170791912) [](ancestors = 170791912,167155809)

Reply:
So can you share what are the diff? I saw many codes the same as in Garry's patch.

---
In reply to: [172435535](https://github.com/Microsoft/pai/pull/318#discussion_r172435535) [](ancestors = 172435535,170791912,167155809)

Reply:
There are less than 30% common, 70% diff, re-use is impossible :)

---
In reply to: [173097292](https://github.com/Microsoft/pai/pull/318#discussion_r173097292) [](ancestors = 173097292,172435535,170791912,167155809)